concent_api/core/models.py

```diff
@@ -497,13 +497,15 @@ def clean(self) -> None:
             deserialized_report_computed_task = deserialize_message(self.report_computed_task.data.tobytes())  # pylint: disable=no-member
 
         # If available, the report_computed_task nested in force_get_task_result must match report_computed_task.
-        if (
-            self.force_get_task_result is not None and
-            deserialize_message(self.force_get_task_result.data).report_computed_task != deserialized_report_computed_task
-        ):
-            raise ValidationError({
-                'force_get_task_result': "ReportComputedTask nested in ForceGetTaskResult must match Subtask's ReportComputedTask."
-            })
+        if self.force_get_task_result is not None:
```

Comment:
Please look at the comment above. In your version it does not match to new block of code in my opinion. 
Second thing- I know that `clean` is very long method now and you did it similar to the rest, but please extract your new block of code to one inner method (e.g. `_deserialize_force_get_task_result`) and do the whole logic in this method.

Reply:
Also, in previous version, if `self.force_get_task_result` was in fact `None`, we've raised `ValidationError`. Now it's just pass, is that a bug or we now want to allow `None` value?

Reply:
@pawelkisielewicz 
> Please look at the comment above. In your version it does not match to new block of code in my opinion.

This block of code has same rules of working like before. Only one difference in this is that we can deserialize it if `self.force_get_task_result.data` is database object instead of string of bytes.

>Second thing- I know that clean is very long method now and you did it similar to the rest, but please extract your new block of code to one inner method (e.g. _deserialize_force_get_task_result) and do the whole logic in this method.

I think that this issue and this branch is not good place for refactor like this. It could be added new issue for refactor `clean()` method in `Subtask` model. If we want and we have time for this we can refactor whole `Subtask` not just a part of it.

@Jakub89 
>Also, in previous version, if self.force_get_task_result was in fact None, we've raised ValidationError. Now it's just pass, is that a bug or we now want to allow None value?

Previously we raised `ValidationError` only when `self.force_get_task_result is not None` **and** nested `report_computed_task of correctly deserialized message was different than `report_computed_task` stored separately. This block of code still works the same:
- we check if `self.force_get_task_result` exists
- try to deserialize it
- compare `self.force_get_task_results.report_computed_task` with `report_computed_task` of this same `Subtask` but stored separately and if it's not the same we raise `ValidationError`

Reply:
>I think that this issue and this branch is not good place for refactor like this. It could be added new issue for refactor clean() method in Subtask model. If we want and we have time for this we can refactor whole Subtask not just a part of it.
I disagree, extraction some code (which is changed by You) to one method is not refactor. If You are touching some code and there is possibility to leave it better I think You should do it.